clang/include/clang/Basic/Builtins.h

```diff
@@ -112,6 +112,16 @@ static constexpr std::array<Info, N> MakeInfos(std::array<Info, N> Infos) {
   return Infos;
 }
 
+/// A shard of a target's builtins string table and info.
+///
+/// Target builtins are sharded across multiple tables due to different
+/// structures, origins, and also to improve the overall scaling by avoiding a
+/// single table across all builtins.
+struct InfosShard {
+  const llvm::StringTable *Strings;
+  llvm::ArrayRef<Info> Infos;
```

Comment:
Huh... looks like this happened? I presume/hope this gets used everywhere we used a std::pair in the previous patch.